.github/workflows/ci_windows_x64_msvc.yml

```diff
@@ -45,8 +46,16 @@ jobs:
       - name: "Installing requirements"
         run: choco install ccache --yes
       - name: "Configuring MSVC"
-        uses: ilammy/msvc-dev-cmd@0b201ec74fa43914dc39ae48a89fd1d8cb592756 # v1.13.0
-      - name: "Building IREE"
-        run: ./build_tools/cmake/build_all.sh "${BUILD_DIR}"
+        uses: ilammy/msvc-dev-cmd@v1.13.0
+      - name: "Installing iree reqs"
+        run: |
+          choco install cmake
+          choco install ninja
+          Import-Module $env:ChocolateyInstall\helpers\chocolateyProfile.psm1
+          refreshenv
+      - name: "Building Iree"
+        run: bash ./build_tools/cmake/build_all.sh "/c/mnt/azure/build-windows"
       - name: "Testing IREE"
-        run: ./build_tools/cmake/ctest_all.sh "${BUILD_DIR}"
+        run: bash ./build_tools/cmake/ctest_all.sh "/c/mnt/azure/build-windows"
+      - name: "Clean up build dir"
+        run: Remove-Item -Path "C:\mnt\azure\build-windows" -Recurse -Force
```

Comment:
yes

Reply:
Although, I have it run at the beginning because if the workflow fails, it doesn't reach the cleanup step. Do you know how to add this to the process that runs after the workflow is completed, regardless of if it succeeds or not?

Reply:
Why? I thought our Linux cluster had ephemeral runners? Wouldn't the Windows cluster follow the same convention? (All of this should go on the tracking issue, PR description, etc. - I'm reading between the lines here when it would be much easier to review the design if it was documented)

Reply:
> Do you know how to add this to the process that runs after the workflow is completed, regardless of if it succeeds or not?

In the workflow file there are options: https://stackoverflow.com/questions/58858429/how-to-run-a-github-actions-step-even-if-the-previous-step-fails-while-still-f

For this sort of self-hosted runner setup though, it's usually safer to configure at the runner level, like https://github.com/iree-org/iree/blob/main/build_tools/github_actions/runner/config/hooks/post_job.sh (maybe ARC has a similar hook you can connect to)

Reply:
@saienduri or @Eliasj42 can either of you comment on persistent vs ephemeral runner usage here? We can start with the "clean up build dir" workflow steps, but longer term I'd like to see either:

1. Ephemeral runners that scale up/down, with no need for explicit clean up steps (the entire disk will go away)
2. Persistent runners that use a pre-job or post-job script configured at the runner / Docker(?) level that hides this detail from workflows

Users should be able to work with these runners without needing to understand the details of file paths, storage mounts, etc. As we add more workflows using these runners, maintaining such steps across each workflow and project will be difficult.

Reply:
These are still ephemeral runners that scale up and down based on github activity. The runners that are spun up use the same storage mount that we clean up after every run due to some storage problems with the ones that the runners are coming up with. Because we are completely clearing the storage every time, this should be fine.

Reply:
Oh, the OS partition is ephemeral but the storage is persistent? Do multiple runners share the same storage disk then...? Can you elaborate on "some storage problems" somewhere (GitHub tracking issue would be best, but I'll settle for a PR review thread like here or one of our internal chats) so we can investigate?

Reply:
This was initially for the nightly case, but yes, as we scale this up and move it to presubmit, we should be moving to an ephemeral storage mount which is pretty easy to do in kubernetes (in which case we remove the cleanup step as well). @Eliasj42 has more context on the issues with the storage problems, but I remember he tried an exhaustive list of things over a couple of days which is when we decided to pivot to a mount

Reply:
Okay, but we'll be aiming for presubmit (`pull_request`) or at least postsubmit (`push`). Nightly already works today and is free, so this would "just" be speeding up that build and taking on some extra billing costs if we stop after just this PR.

Reply:
Postsubmit with a 1 hour build on a single runner may be reasonable to start putting mileage on this.